src/ripple/protocol/impl/SField.cpp

```diff
@@ -184,6 +188,7 @@ CONSTRUCT_TYPED_SFIELD(sfTakerPaysCurrency,     "TakerPaysCurrency",    UINT160,
 CONSTRUCT_TYPED_SFIELD(sfTakerPaysIssuer,       "TakerPaysIssuer",      UINT160,    2);
 CONSTRUCT_TYPED_SFIELD(sfTakerGetsCurrency,     "TakerGetsCurrency",    UINT160,    3);
 CONSTRUCT_TYPED_SFIELD(sfTakerGetsIssuer,       "TakerGetsIssuer",      UINT160,    4);
+CONSTRUCT_TYPED_SFIELD(sfAssetCode,             "AssetCode",            UINT160,    5);
```

Comment:
One annoying thing about `AssetCode` being typed as a `UINT160` is that in JSON, they're never represented as ISO codes. ie if you're CFT's asset code is "USD", in JSON it'll be represented as `"0000000000000000000000005553440000000000"`. Might be nice to have some custom logic in wherever UINT160s are serialized to JSON to serialize to ISO format if applicable.

Reply:
Or, we could not use UInt160s to represent assets codes, at least for this use case.  See a different comment that explores using a UInt64 to store ISO-like codes of up to eight ASCII characters.

Reply:
honestly - i'm in favor of keeping this part the same as IOUs from a user perspective: IE currency codes are 100% compatible between the two. why? because we can and it provides a benefit to maintain backwards compatibility where possible and because the space-saving here is likely to be trivial, as there are not likely to be many many CFTs. 

yes, it's true that this value still has to be encoded in the STAmount, so there it will also have a space penalty, but we're not making anything worse - STAmount size for IOU and CFT will be the same

furthermore, I don't think that rippled should make technical decisions based on the convenience of library writers :) . IE, the place for a ISO -> internal translation should be w/in the library. I see that as a within the bounds of what a library should do.

Reply:
This is great feedback - mind sharing it in the discussion I started here: https://github.com/XRPLF/XRPL-Standards/discussions/128

(if so, this comment can probably be resolved in-favor of that discussion -- we could introduce a new PR if anything changes out of that discussion).

Reply:
@sappenin @scottschurr and i have agreed not to change the currency code format. the best reason to do so would be for space-saving, but an upcoming change in the CFT ID will make this moot